docs/components/concepts/user-task-listeners.md

```diff
@@ -0,0 +1,179 @@
+
+### Correcting user task data
+
+User task listeners can correct user task data before the lifecycle transition is finalized. Corrections allow user task listeners to update specific attributes of the user task, such as the assignee, due date, follow-up date, candidate users, candidate groups, and priority. These corrections are applied after all task listeners for the current operation have successfully completed.
+
+If an operation is denied by a listener, no corrections are applied to the user task.
+
+Below is an example of how to correct the user task data form a job worker while completing the user task listener job:
+
+```java
+final JobHandler completeTaskListenerJobWithCorrectionsHandler =
+    (jobClient, job) ->
+        jobClient
+            .newCompleteCommand(job)
+            // highlight-start
+            .withResult(
+                new CompleteJobResult()
+                    .correctAssignee("john_doe") // assigns the user task to 'john_doe'
+                    .correctDueDate(null) // preserves the current 'dueDate' of the user task
+                    .correctFollowUpDate("") // clears the 'followUpDate'
+                    .correctCandidateUsers(List.of("alice", "bob")) // sets candidate users
+                    .correctCandidateGroups(List.of()) // clears the candidate groups
+                    .correctPriority(80)) // sets the priority to 80
+            // highlight-end
+            .send();
+
+client.newWorker()
+    .jobType("user-task-listener-completion") // type of the user task listener job
+    .handler(completeTaskListenerJobWithCorrectionsHandler)
+    .open();
+```
+
+### Denying the operation
+
+User task listeners can deny a user task operation, such as assignment or completion, effectively preventing the lifecycle transition from completing.
+
+When an operation is denied:
+
+- **Corrections discarded**: Any corrections made by preceding listeners within the same operation are discarded.
+- **Task state preserved**: The user task retains its state and data as if the operation was never attempted.
+
+This capability is particularly useful for implementing validation logic or enforcing business rules before allowing a user task operation to proceed.
+
+Below is an example of how to deny a user task operation from a job worker while completing the user task listener job:
+
+```java
+final JobHandler denyUserTaskOperationHandler =
+    (jobClient, job) ->
+        jobClient
+            .newCompleteCommand(job)
+            // highlight-start
+            .withResult()
+            .deny(true)
+            // highlight-end
+            .send();
+```
+
+## Expression evaluation and incident behavior
+
+### Expression evaluation
+
+User task listener properties, such as job `type` or `retries`, are evaluated right before the job creation for the listener. If an expression evaluation fails, the operation is paused, and an incident is raised.
```

Comment:
perhaps link to https://docs.camunda.io/docs/components/concepts/incidents/ here?

Reply:
Although I see it is listed in related resources. Up to you.